components/community-groups/CommunityGroups.jsx

```diff
@@ -137,6 +194,12 @@ class CommunityGroups extends React.Component {
           </NativeSelect>
         <FormHelperText>Filter by gender</FormHelperText>
         </FormControl>
+        <Button className={classes.newCGbutton} variant="contained" 
+        color="primary" onClick={() => this.handleForm(true)}>
+          New CG
+        </Button>
+        {!this.state.loading && addCGform}

```

Comment:
This ensures that loading has been done before we can allow addCGform to be rendered. This is necessary because addCGform receives the data from states which are retrieved from Firestore. If we don't wait on loading, React automatically renders addCGform which breaks the entire app since there states don't have the expected data.